src/revwalk.c

```diff
@@ -398,81 +398,191 @@ static int revwalk_next_reverse(git_commit_list_node **object_out, git_revwalk *
 	return *object_out ? 0 : GIT_ITEROVER;
 }
 
-
-static int interesting(git_pqueue *list)
+static int contains(git_pqueue *list, git_commit_list_node *node)
```

Comment:
This looks like `git_vector_search`, which should in fact be more efficient. So maybe just add `#define git_pqueue_search git_pqueue_search` and remove this function altogether?